src/linker/Linker.Dataflow/DynamicallyAccessedMembersTypeHierarchy.cs

```diff
@@ -106,8 +108,9 @@ public DynamicallyAccessedMembersTypeHierarchy (LinkContext context, MarkStep ma
 			if (apply) {
 				// One of the base/interface types is already marked as having the annotation applied
 				// so we need to apply the annotation to this type as well
-				var reflectionMethodBodyScanner = new ReflectionMethodBodyScanner (_context, _markStep);
-				var reflectionPatternContext = new ReflectionPatternContext (_context, true, type, type);
+				using var _ = _scopeStack.PushScope (new MessageOrigin (type));
+				var reflectionMethodBodyScanner = new ReflectionMethodBodyScanner (_context, _markStep, _scopeStack);
+				var reflectionPatternContext = new ReflectionPatternContext (_context, true, _scopeStack.CurrentScope.Origin, type);
```

Comment:
Would it make sense to add overload with ReflectionPatternContext parameter?

Reply:
Sorry - I don't follow. Overload of which function call here?

Reply:
Of the constructors or make static factory methods. Something like this 

for types
```
reflectionPatternContext = ReflectionPatternContext.ForType (_context, _scopeStack, type);
```
for methods
```
reflectionPatternContext = ReflectionPatternContext.ForMethod (_context, _scopeStack, method);
```

Reply:
Definitely possible, but I don't see the benefits. There's nothing different between methods/types/fields in this case. The only advantage would be to potentially restrict what can be passed in a bit more (not just random "member", but "method/type/field" only).

Reply:
Fair enough, it was just a suggestion